diff -Naur pamac-6.0.2-orig/resources/preferences_dialog.ui pamac-6.0.2/resources/preferences_dialog.ui
--- pamac-6.0.2-orig/resources/preferences_dialog.ui	2017-09-22 07:14:57.000000000 -0300
+++ pamac-6.0.2/resources/preferences_dialog.ui	2017-09-24 18:12:39.296000000 -0300
@@ -518,193 +518,6 @@
               </packing>
             </child>
             <child>
-              <object class="GtkBox" id="aur_config_box">
-                <property name="visible">True</property>
-                <property name="can_focus">False</property>
-                <property name="orientation">vertical</property>
-                <property name="spacing">6</property>
-                <child>
-                  <object class="GtkInfoBar" id="aur_infobar">
-                    <property name="visible">True</property>
-                    <property name="app_paintable">True</property>
-                    <property name="can_focus">False</property>
-                    <property name="message_type">warning</property>
-                    <child internal-child="action_area">
-                      <object class="GtkButtonBox" id="infobar-action_area">
-                        <property name="can_focus">False</property>
-                        <property name="spacing">6</property>
-                        <child>
-                          <placeholder/>
-                        </child>
-                      </object>
-                      <packing>
-                        <property name="expand">False</property>
-                        <property name="fill">False</property>
-                        <property name="position">0</property>
-                      </packing>
-                    </child>
-                    <child internal-child="content_area">
-                      <object class="GtkBox" id="infobar-content_area">
-                        <property name="can_focus">False</property>
-                        <property name="spacing">16</property>
-                        <child>
-                          <object class="GtkLabel" id="aur_warning_label">
-                            <property name="visible">True</property>
-                            <property name="can_focus">False</property>
-                            <property name="halign">start</property>
-                            <property name="hexpand">True</property>
-                            <property name="label" translatable="yes">AUR is a community maintained repository so it presents potential risks and problems.
-All AUR users should be familiar with the build process.</property>
-                            <property name="wrap">True</property>
-                            <property name="max_width_chars">300</property>
-                          </object>
-                          <packing>
-                            <property name="expand">False</property>
-                            <property name="fill">True</property>
-                            <property name="position">0</property>
-                          </packing>
-                        </child>
-                      </object>
-                      <packing>
-                        <property name="expand">False</property>
-                        <property name="fill">False</property>
-                        <property name="position">0</property>
-                      </packing>
-                    </child>
-                    <child>
-                      <placeholder/>
-                    </child>
-                  </object>
-                  <packing>
-                    <property name="expand">False</property>
-                    <property name="fill">True</property>
-                    <property name="position">0</property>
-                  </packing>
-                </child>
-                <child>
-                  <object class="GtkBox" id="box2">
-                    <property name="visible">True</property>
-                    <property name="can_focus">False</property>
-                    <property name="spacing">12</property>
-                    <child>
-                      <object class="GtkLabel" id="enable_aur_label">
-                        <property name="visible">True</property>
-                        <property name="can_focus">False</property>
-                        <property name="halign">start</property>
-                        <property name="margin_left">6</property>
-                        <property name="margin_right">6</property>
-                        <property name="margin_start">6</property>
-                        <property name="margin_end">6</property>
-                        <property name="hexpand">True</property>
-                        <property name="label" translatable="yes">Enable AUR support</property>
-                      </object>
-                      <packing>
-                        <property name="expand">False</property>
-                        <property name="fill">True</property>
-                        <property name="position">0</property>
-                      </packing>
-                    </child>
-                    <child>
-                      <object class="GtkSwitch" id="enable_aur_button">
-                        <property name="visible">True</property>
-                        <property name="can_focus">True</property>
-                        <property name="tooltip_text" translatable="yes">Allow Pamac to search and install packages from AUR</property>
-                        <property name="halign">end</property>
-                        <property name="margin_left">6</property>
-                        <property name="margin_right">6</property>
-                        <property name="margin_start">6</property>
-                        <property name="margin_end">6</property>
-                      </object>
-                      <packing>
-                        <property name="expand">False</property>
-                        <property name="fill">True</property>
-                        <property name="position">1</property>
-                      </packing>
-                    </child>
-                  </object>
-                  <packing>
-                    <property name="expand">False</property>
-                    <property name="fill">True</property>
-                    <property name="position">1</property>
-                  </packing>
-                </child>
-                <child>
-                  <object class="GtkCheckButton" id="check_aur_updates_checkbutton">
-                    <property name="label" translatable="yes">Check for updates from AUR</property>
-                    <property name="visible">True</property>
-                    <property name="can_focus">True</property>
-                    <property name="receives_default">False</property>
-                    <property name="halign">start</property>
-                    <property name="margin_left">24</property>
-                    <property name="margin_start">24</property>
-                    <property name="hexpand">True</property>
-                    <property name="draw_indicator">True</property>
-                  </object>
-                  <packing>
-                    <property name="expand">False</property>
-                    <property name="fill">True</property>
-                    <property name="position">2</property>
-                  </packing>
-                </child>
-                <child>
-                  <object class="GtkBox" id="box3">
-                    <property name="visible">True</property>
-                    <property name="can_focus">False</property>
-                    <property name="margin_left">26</property>
-                    <property name="spacing">12</property>
-                    <property name="homogeneous">True</property>
-                    <child>
-                      <object class="GtkLabel" id="aur_build_dir_label">
-                        <property name="visible">True</property>
-                        <property name="can_focus">False</property>
-                        <property name="halign">start</property>
-                        <property name="hexpand">True</property>
-                      </object>
-                      <packing>
-                        <property name="expand">False</property>
-                        <property name="fill">True</property>
-                        <property name="position">0</property>
-                      </packing>
-                    </child>
-                    <child>
-                      <object class="GtkFileChooserButton" id="aur_build_dir_file_chooser">
-                        <property name="visible">True</property>
-                        <property name="can_focus">False</property>
-                        <property name="halign">end</property>
-                        <property name="margin_left">24</property>
-                        <property name="action">select-folder</property>
-                        <property name="title" translatable="yes"/>
-                      </object>
-                      <packing>
-                        <property name="expand">False</property>
-                        <property name="fill">True</property>
-                        <property name="position">1</property>
-                      </packing>
-                    </child>
-                  </object>
-                  <packing>
-                    <property name="expand">False</property>
-                    <property name="fill">True</property>
-                    <property name="position">3</property>
-                  </packing>
-                </child>
-              </object>
-              <packing>
-                <property name="position">2</property>
-              </packing>
-            </child>
-            <child type="tab">
-              <object class="GtkLabel" id="aur_label">
-                <property name="visible">True</property>
-                <property name="can_focus">False</property>
-                <property name="label" translatable="yes">AUR</property>
-              </object>
-              <packing>
-                <property name="position">2</property>
-                <property name="tab_fill">False</property>
-              </packing>
-            </child>
-            <child>
               <object class="GtkBox" id="cache_config_box">
                 <property name="visible">True</property>
                 <property name="can_focus">False</property>
diff -Naur pamac-6.0.2-orig/src/preferences_dialog.vala pamac-6.0.2/src/preferences_dialog.vala
--- pamac-6.0.2-orig/src/preferences_dialog.vala	2017-09-22 07:14:57.000000000 -0300
+++ pamac-6.0.2/src/preferences_dialog.vala	2017-09-24 18:09:10.462000000 -0300
@@ -47,14 +47,6 @@
 		[GtkChild]
 		Gtk.Button generate_mirrors_list_button;
 		[GtkChild]
-		Gtk.Switch enable_aur_button;
-		[GtkChild]
-		Gtk.Label aur_build_dir_label;
-		[GtkChild]
-		Gtk.FileChooserButton aur_build_dir_file_chooser;
-		[GtkChild]
-		Gtk.CheckButton check_aur_updates_checkbutton;
-		[GtkChild]
 		Gtk.Label cache_keep_nb_label;
 		[GtkChild]
 		Gtk.SpinButton cache_keep_nb_spin_button;
@@ -73,7 +65,6 @@
 			this.transaction = transaction;
 			refresh_period_label.set_markup (dgettext (null, "How often to check for updates, value in hours") +":");
 			cache_keep_nb_label.set_markup (dgettext (null, "Number of versions of each package to keep in the cache") +":");
-			aur_build_dir_label.set_markup (dgettext (null, "Build directory") +":");
 			remove_unrequired_deps_button.active = transaction.recurse;
 			check_space_button.active = transaction.get_checkspace ();
 			if (transaction.refresh_period == 0) {
@@ -139,22 +130,6 @@
 				mirrors_list_generation_method_comboboxtext.changed.connect (on_mirrors_list_generation_method_comboboxtext_changed);
 				transaction.write_mirrors_config_finished.connect (on_write_mirrors_config_finished);
 			}
-
-			enable_aur_button.active = transaction.enable_aur;
-			aur_build_dir_label.sensitive = transaction.enable_aur;
-			aur_build_dir_file_chooser.sensitive = transaction.enable_aur;
-			aur_build_dir_file_chooser.set_filename (transaction.aur_build_dir);
-			// add /tmp choice always visible
-			try {
-				aur_build_dir_file_chooser.add_shortcut_folder ("/tmp");
-			} catch (GLib.Error e) {
-				stderr.printf ("%s\n", e.message);
-			}
-			check_aur_updates_checkbutton.active = transaction.check_aur_updates;
-			check_aur_updates_checkbutton.sensitive = transaction.enable_aur;
-			enable_aur_button.state_set.connect (on_enable_aur_button_state_set);
-			aur_build_dir_file_chooser.file_set.connect (on_aur_build_dir_set);
-			check_aur_updates_checkbutton.toggled.connect (on_check_aur_updates_checkbutton_toggled);
 		}
 
 		bool on_remove_unrequired_deps_button_state_set (bool new_state) {
@@ -203,27 +178,7 @@
 			transaction.start_write_pamac_config (new_pamac_conf);
 		}
 
-		bool on_enable_aur_button_state_set (bool new_state) {
-			var new_pamac_conf = new HashTable<string,Variant> (str_hash, str_equal);
-			new_pamac_conf.insert ("EnableAUR", new Variant.boolean (new_state));
-			transaction.start_write_pamac_config (new_pamac_conf);
-			return true;
-		}
-
-		void on_aur_build_dir_set () {
-			var new_pamac_conf = new HashTable<string,Variant> (str_hash, str_equal);
-			new_pamac_conf.insert ("BuildDirectory", new Variant.string (aur_build_dir_file_chooser.get_filename ()));
-			transaction.start_write_pamac_config (new_pamac_conf);
-		}
-
-		void on_check_aur_updates_checkbutton_toggled () {
-			var new_pamac_conf = new HashTable<string,Variant> (str_hash, str_equal);
-			new_pamac_conf.insert ("CheckAURUpdates", new Variant.boolean (check_aur_updates_checkbutton.active));
-			transaction.start_write_pamac_config (new_pamac_conf);
-		}
-
-		void on_write_pamac_config_finished (bool recurse, uint64 refresh_period, bool no_update_hide_icon,
-											bool enable_aur, string aur_build_dir, bool check_aur_updates) {
+		void on_write_pamac_config_finished (bool recurse, uint64 refresh_period, bool no_update_hide_icon) {
 			remove_unrequired_deps_button.state = recurse;
 			if (refresh_period == 0) {
 				check_updates_button.state = false;
@@ -241,11 +196,6 @@
 				ignorepkgs_box.sensitive = true;
 			}
 			no_update_hide_icon_checkbutton.active = no_update_hide_icon;
-			enable_aur_button.state = enable_aur;
-			aur_build_dir_label.sensitive = enable_aur;
-			aur_build_dir_file_chooser.sensitive = enable_aur;
-			check_aur_updates_checkbutton.active = check_aur_updates;
-			check_aur_updates_checkbutton.sensitive = enable_aur;
 		}
 
 		bool on_check_space_button_state_set (bool new_state) {
