diff -Naur pamac-5.1.1-orig/resources/preferences_dialog.ui pamac-5.1.1/resources/preferences_dialog.ui
--- pamac-5.1.1-orig/resources/preferences_dialog.ui	2017-09-07 03:44:43.000000000 -0300
+++ pamac-5.1.1/resources/preferences_dialog.ui	2017-09-12 21:03:46.784000000 -0300
@@ -491,161 +491,6 @@
               </packing>
             </child>
             <child>
-              <object class="GtkBox" id="aur_config_box">
-                <property name="visible">True</property>
-                <property name="can_focus">False</property>
-                <property name="orientation">vertical</property>
-                <property name="spacing">6</property>
-                <child>
-                  <object class="GtkInfoBar" id="aur_infobar">
-                    <property name="visible">True</property>
-                    <property name="app_paintable">True</property>
-                    <property name="can_focus">False</property>
-                    <property name="message_type">warning</property>
-                    <child internal-child="action_area">
-                      <object class="GtkButtonBox" id="infobar-action_area">
-                        <property name="can_focus">False</property>
-                        <property name="spacing">6</property>
-                        <child>
-                          <placeholder/>
-                        </child>
-                      </object>
-                      <packing>
-                        <property name="expand">False</property>
-                        <property name="fill">False</property>
-                        <property name="position">0</property>
-                      </packing>
-                    </child>
-                    <child internal-child="content_area">
-                      <object class="GtkBox" id="infobar-content_area">
-                        <property name="can_focus">False</property>
-                        <property name="spacing">16</property>
-                        <child>
-                          <object class="GtkLabel" id="aur_warning_label">
-                            <property name="visible">True</property>
-                            <property name="can_focus">False</property>
-                            <property name="halign">start</property>
-                            <property name="hexpand">True</property>
-                            <property name="label" translatable="yes">AUR is a community maintained repository so it presents potential risks and problems.
-All AUR users should be familiar with the build process.</property>
-                            <property name="wrap">True</property>
-                            <property name="max_width_chars">300</property>
-                          </object>
-                          <packing>
-                            <property name="expand">False</property>
-                            <property name="fill">True</property>
-                            <property name="position">0</property>
-                          </packing>
-                        </child>
-                      </object>
-                      <packing>
-                        <property name="expand">False</property>
-                        <property name="fill">False</property>
-                        <property name="position">0</property>
-                      </packing>
-                    </child>
-                    <child>
-                      <placeholder/>
-                    </child>
-                  </object>
-                  <packing>
-                    <property name="expand">False</property>
-                    <property name="fill">True</property>
-                    <property name="position">0</property>
-                  </packing>
-                </child>
-                <child>
-                  <object class="GtkBox" id="box2">
-                    <property name="visible">True</property>
-                    <property name="can_focus">False</property>
-                    <property name="spacing">12</property>
-                    <property name="homogeneous">True</property>
-                    <child>
-                      <object class="GtkLabel" id="enable_aur_label">
-                        <property name="visible">True</property>
-                        <property name="can_focus">False</property>
-                        <property name="margin_left">6</property>
-                        <property name="margin_right">6</property>
-                        <property name="margin_start">6</property>
-                        <property name="margin_end">6</property>
-                        <property name="hexpand">True</property>
-                        <property name="label" translatable="yes">Enable AUR support</property>
-                        <property name="xalign">0</property>
-                      </object>
-                      <packing>
-                        <property name="expand">False</property>
-                        <property name="fill">True</property>
-                        <property name="position">0</property>
-                      </packing>
-                    </child>
-                    <child>
-                      <object class="GtkSwitch" id="enable_aur_button">
-                        <property name="visible">True</property>
-                        <property name="can_focus">True</property>
-                        <property name="tooltip_text" translatable="yes">Allow Pamac to search and install packages from AUR</property>
-                        <property name="halign">end</property>
-                        <property name="margin_left">6</property>
-                        <property name="margin_right">6</property>
-                        <property name="margin_start">6</property>
-                        <property name="margin_end">6</property>
-                      </object>
-                      <packing>
-                        <property name="expand">False</property>
-                        <property name="fill">True</property>
-                        <property name="position">1</property>
-                      </packing>
-                    </child>
-                  </object>
-                  <packing>
-                    <property name="expand">False</property>
-                    <property name="fill">True</property>
-                    <property name="position">1</property>
-                  </packing>
-                </child>
-                <child>
-                  <object class="GtkCheckButton" id="search_aur_checkbutton">
-                    <property name="label" translatable="yes">Search in AUR by default</property>
-                    <property name="visible">True</property>
-                    <property name="can_focus">True</property>
-                    <property name="receives_default">False</property>
-                    <property name="halign">start</property>
-                    <property name="margin_left">24</property>
-                    <property name="margin_start">24</property>
-                    <property name="hexpand">True</property>
-                    <property name="draw_indicator">True</property>
-                  </object>
-                  <packing>
-                    <property name="expand">False</property>
-                    <property name="fill">True</property>
-                    <property name="position">2</property>
-                  </packing>
-                </child>
-                <child>
-                  <object class="GtkCheckButton" id="check_aur_updates_checkbutton">
-                    <property name="label" translatable="yes">Check for updates from AUR</property>
-                    <property name="visible">True</property>
-                    <property name="can_focus">True</property>
-                    <property name="receives_default">False</property>
-                    <property name="halign">start</property>
-                    <property name="margin_left">24</property>
-                    <property name="margin_start">24</property>
-                    <property name="hexpand">True</property>
-                    <property name="draw_indicator">True</property>
-                  </object>
-                  <packing>
-                    <property name="expand">False</property>
-                    <property name="fill">True</property>
-                    <property name="position">3</property>
-                  </packing>
-                </child>
-              </object>
-              <packing>
-                <property name="name">aur</property>
-                <property name="title" translatable="yes">AUR</property>
-                <property name="position">2</property>
-              </packing>
-            </child>
-            <child>
               <object class="GtkBox" id="cache_config_box">
                 <property name="visible">True</property>
                 <property name="can_focus">False</property>
diff -Naur pamac-5.1.1-orig/src/preferences_dialog.vala pamac-5.1.1/src/preferences_dialog.vala
--- pamac-5.1.1-orig/src/preferences_dialog.vala	2017-09-07 03:44:43.000000000 -0300
+++ pamac-5.1.1/src/preferences_dialog.vala	2017-09-12 21:11:18.886000000 -0300
@@ -47,12 +47,6 @@
 		[GtkChild]
 		Gtk.Button generate_mirrors_list_button;
 		[GtkChild]
-		Gtk.Switch enable_aur_button;
-		[GtkChild]
-		Gtk.CheckButton search_aur_checkbutton;
-		[GtkChild]
-		Gtk.CheckButton check_aur_updates_checkbutton;
-		[GtkChild]
 		Gtk.Label cache_keep_nb_label;
 		[GtkChild]
 		Gtk.SpinButton cache_keep_nb_spin_button;
@@ -129,15 +123,6 @@
 				mirrors_list_generation_method_comboboxtext.changed.connect (on_mirrors_list_generation_method_comboboxtext_changed);
 				transaction.write_mirrors_config_finished.connect (on_write_mirrors_config_finished);
 			}
-
-			enable_aur_button.active = transaction.enable_aur;
-			search_aur_checkbutton.active = transaction.search_aur;
-			search_aur_checkbutton.sensitive = transaction.enable_aur;
-			check_aur_updates_checkbutton.active = transaction.check_aur_updates;
-			check_aur_updates_checkbutton.sensitive = transaction.enable_aur;
-			enable_aur_button.state_set.connect (on_enable_aur_button_state_set);
-			search_aur_checkbutton.toggled.connect (on_search_aur_checkbutton_toggled);
-			check_aur_updates_checkbutton.toggled.connect (on_check_aur_updates_checkbutton_toggled);
 		}
 
 		bool on_remove_unrequired_deps_button_state_set (bool new_state) {
@@ -174,25 +159,6 @@
 			transaction.start_write_pamac_config (new_pamac_conf);
 		}
 
-		bool on_enable_aur_button_state_set (bool new_state) {
-			var new_pamac_conf = new HashTable<string,Variant> (str_hash, str_equal);
-			new_pamac_conf.insert ("EnableAUR", new Variant.boolean (new_state));
-			transaction.start_write_pamac_config (new_pamac_conf);
-			return true;
-		}
-
-		void on_search_aur_checkbutton_toggled () {
-			var new_pamac_conf = new HashTable<string,Variant> (str_hash, str_equal);
-			new_pamac_conf.insert ("SearchInAURByDefault", new Variant.boolean (search_aur_checkbutton.active));
-			transaction.start_write_pamac_config (new_pamac_conf);
-		}
-
-		void on_check_aur_updates_checkbutton_toggled () {
-			var new_pamac_conf = new HashTable<string,Variant> (str_hash, str_equal);
-			new_pamac_conf.insert ("CheckAURUpdates", new Variant.boolean (check_aur_updates_checkbutton.active));
-			transaction.start_write_pamac_config (new_pamac_conf);
-		}
-
 		void on_write_pamac_config_finished (bool recurse, uint64 refresh_period, bool no_update_hide_icon,
 											bool enable_aur, bool search_aur, bool check_aur_updates) {
 			remove_unrequired_deps_button.state = recurse;
@@ -212,11 +178,6 @@
 				ignorepkgs_box.sensitive = true;
 			}
 			no_update_hide_icon_checkbutton.active = no_update_hide_icon;
-			enable_aur_button.state = enable_aur;
-			search_aur_checkbutton.active = search_aur;
-			search_aur_checkbutton.sensitive = enable_aur;
-			check_aur_updates_checkbutton.active = check_aur_updates;
-			check_aur_updates_checkbutton.sensitive = enable_aur;
 		}
 
 		bool on_check_space_button_state_set (bool new_state) {
