diff -Naur openrc-orig/init.d/agetty.in openrc/init.d/agetty.in
--- openrc-orig/init.d/agetty.in	2017-06-14 21:52:33.451000000 -0400
+++ openrc/init.d/agetty.in	2017-06-14 21:36:43.218000000 -0400
@@ -16,6 +16,7 @@
 command=/sbin/agetty
 command_args_foreground="${agetty_options} ${port} ${baud} ${term_type}"
 pidfile="/run/${RC_SVCNAME}.pid"
+quiet="YES"
 
 depend() {
 	after local
diff -Naur openrc-orig/sh/start-stop-daemon.sh openrc/sh/start-stop-daemon.sh
--- openrc-orig/sh/start-stop-daemon.sh	2017-06-14 21:52:33.456000000 -0400
+++ openrc/sh/start-stop-daemon.sh	2017-06-14 22:59:05.249000000 -0400
@@ -21,14 +21,20 @@
 	fi
 
 	local _background=
-	ebegin "Starting ${name:-$RC_SVCNAME}"
+	if [ ! "${quiet}" == "YES" ]; then
+		ebegin "Starting ${name:-$RC_SVCNAME}"
+	fi
 	if yesno "${command_background}"; then
 		if [ -z "${pidfile}" ]; then
-			eend 1 "command_background option used but no pidfile specified"
+			if [ ! "${quiet}" == "YES" ]; then
+				eend 1 "command_background option used but no pidfile specified"
+			fi
 			return 1
 		fi
 		if [ -n "${command_args_background}" ]; then
-			eend 1 "command_background used with command_args_background"
+			if [ ! "${quiet}" == "YES" ]; then
+				eend 1 "command_background used with command_args_background"
+			fi
 			return 1
 		fi
 		_background="--background --make-pidfile"
@@ -49,7 +55,11 @@
 		${command_user+--user} $command_user \
 		$_background $start_stop_daemon_args \
 		-- $command_args $command_args_background
-	if eend $? "Failed to start ${name:-$RC_SVCNAME}"; then
+	rc=$?
+	if [ ! "${quiet}" == "YES" ]; then
+		eend $rc "Failed to start ${name:-$RC_SVCNAME}"
+	fi
+	if [ $rc ]; then
 		service_set_value "command" "${command}"
 		[ -n "${chroot}" ] && service_set_value "chroot" "${chroot}"
 		[ -n "${pidfile}" ] && service_set_value "pidfile" "${pidfile}"
@@ -77,7 +87,9 @@
 	procname="${startprocname:-$procname}"
 	[ -n "$command" -o -n "$procname" -o -n "$pidfile" ] || return 0
 	yesno "${command_progress}" && _progress=--progress
-	ebegin "Stopping ${name:-$RC_SVCNAME}"
+	if [ ! "${quiet}" == "YES" ]; then
+		ebegin "Stopping ${name:-$RC_SVCNAME}"
+	fi
 	start-stop-daemon --stop \
 		${retry:+--retry} $retry \
 		${command:+--exec} $command \
@@ -86,7 +98,9 @@
 		${stopsig:+--signal} $stopsig \
 		${_progress}
 
-	eend $? "Failed to stop ${name:-$RC_SVCNAME}"
+	if [ ! "${quiet}" == "YES" ]; then
+		eend $? "Failed to stop ${name:-$RC_SVCNAME}"
+	fi
 }
 
 ssd_status()
diff -Naur openrc-orig/sh/supervise-daemon.sh openrc/sh/supervise-daemon.sh
--- openrc-orig/sh/supervise-daemon.sh	2017-06-14 21:52:33.456000000 -0400
+++ openrc/sh/supervise-daemon.sh	2017-06-14 22:59:30.953000000 -0400
@@ -18,7 +18,9 @@
 		return 1
 	fi
 
-	ebegin "Starting ${name:-$RC_SVCNAME}"
+	if [ ! "${quiet}" == "YES" ]; then
+		ebegin "Starting ${name:-$RC_SVCNAME}"
+	fi
 	# The eval call is necessary for cases like:
 	# command_args="this \"is a\" test"
 	# to work properly.
@@ -37,7 +39,9 @@
 		[ -n "${chroot}" ] && service_set_value "chroot" "${chroot}"
 		[ -n "${pidfile}" ] && service_set_value "pidfile" "${pidfile}"
 	fi
-	eend $rc "failed to start ${name:-$RC_SVCNAME}"
+	if [ ! "${quiet}" == "YES" ]; then
+		eend $rc "failed to start ${name:-$RC_SVCNAME}"
+	fi
 }
 
 supervise_stop()
@@ -47,12 +51,16 @@
 	chroot="${startchroot:-$chroot}"
 	pidfile="${startpidfile:-$pidfile}"
 	[ -n "$pidfile" ] || return 0
-	ebegin "Stopping ${name:-$RC_SVCNAME}"
+	if [ ! "${quiet}" == "YES" ]; then
+		ebegin "Stopping ${name:-$RC_SVCNAME}"
+	fi
 	supervise-daemon --stop \
 		${pidfile:+--pidfile} $chroot$pidfile \
 		${stopsig:+--signal} $stopsig
 
-	eend $? "Failed to stop ${name:-$RC_SVCNAME}"
+	if [ ! "${quiet}" == "YES" ]; then
+		eend $? "Failed to stop ${name:-$RC_SVCNAME}"
+	fi
 }
 
 supervise_status()
