commit 865c1dda7e13deff04573ffc42028b71fee07f9c
Author: Hubert Figuière <hub@figuiere.net>
Date:   Wed Dec 21 23:29:56 2016 -0500

    Bug 13791 - Fix flickering

diff --git a/src/af/xap/gtk/xap_UnixFrameImpl.cpp b/src/af/xap/gtk/xap_UnixFrameImpl.cpp
index 10f8e00..3292b3d 100644
--- a/src/af/xap/gtk/xap_UnixFrameImpl.cpp
+++ b/src/af/xap/gtk/xap_UnixFrameImpl.cpp
@@ -1215,8 +1215,16 @@ gboolean XAP_UnixFrameImpl::_fe::expose(GtkWidget * w, GdkEventExpose* pExposeEv
 {
 	XAP_UnixFrameImpl * pUnixFrameImpl = static_cast<XAP_UnixFrameImpl *>(g_object_get_data(G_OBJECT(w), "user_data"));
 	FV_View * pView = static_cast<FV_View *>(pUnixFrameImpl->getFrame()->getCurrentView());
+	double x, y, width, height;
 #if GTK_CHECK_VERSION(3,0,0)
-	GdkEventExpose *pExposeEvent = reinterpret_cast<GdkEventExpose *>(gtk_get_current_event());
+	cairo_clip_extents (cr, &x, &y, &width, &height);
+	width -= x;
+	height -= y;
+#else
+	x = pExposeEvent->area.x;
+	y = pExposeEvent->area.y;
+	width = pExposeEvent->area.width;
+	height = pExposeEvent->area.height;
 #endif
 /* Jean: commenting out next lines since the zoom update code does draw only
  * part of what needs to be updated. */
@@ -1231,15 +1239,17 @@ gboolean XAP_UnixFrameImpl::_fe::expose(GtkWidget * w, GdkEventExpose* pExposeEv
 		if (pGr->getPaintCount () > 0)
 			return TRUE;
 		xxx_UT_DEBUGMSG(("Expose area: x %d y %d width %d  height %d \n",pExposeEvent->area.x,pExposeEvent->area.y,pExposeEvent->area.width,pExposeEvent->area.height));
-		rClip.left = pGr->tlu(pExposeEvent->area.x);
-		rClip.top = pGr->tlu(pExposeEvent->area.y);
-		rClip.width = pGr->tlu(pExposeEvent->area.width)+1;
-		rClip.height = pGr->tlu(pExposeEvent->area.height)+1;
+		rClip.left = pGr->tlu(x);
+		rClip.top = pGr->tlu(y);
 #if GTK_CHECK_VERSION(3,0,0)
+		rClip.width = pGr->tlu(width);
+		rClip.height = pGr->tlu(height);
 		static_cast<GR_CairoGraphics *>(pGr)->setCairo(cr);
 		pView->draw(&rClip);
 		static_cast<GR_CairoGraphics *>(pGr)->setCairo(NULL);
 #else
+		rClip.width = pGr->tlu(width)+1;
+		rClip.height = pGr->tlu(height)+1;
 		pView->draw(&rClip);
 #endif
 	}
