diff -Naur pacman-5.0.2-orig/src/pacman/conf.c pacman-5.0.2/src/pacman/conf.c
--- pacman-5.0.2-orig/src/pacman/conf.c	2017-04-28 20:18:01.000000000 -0300
+++ pacman-5.0.2/src/pacman/conf.c	2017-09-01 20:03:29.601478215 -0300
@@ -561,6 +561,11 @@
 				config->gpgdir = strdup(value);
 				pm_printf(ALPM_LOG_DEBUG, "config: gpgdir: %s\n", value);
 			}
+		}  else if(strcmp(key, "CodeName") == 0) {
+			if(!config->codename) {
+				config->codename = strdup(value);
+				pm_printf(ALPM_LOG_DEBUG, "config: codename: %s\n", value);
+			}
 		} else if(strcmp(key, "LogFile") == 0) {
 			if(!config->logfile) {
 				config->logfile = strdup(value);
@@ -619,6 +624,7 @@
 	const char *dbname = alpm_db_get_name(db);
 	/* let's attempt a replacement for the current repo */
 	char *temp = strreplace(value, "$repo", dbname);
+	temp = strreplace(temp, "$codename", config->codename);
 	/* let's attempt a replacement for the arch */
 	const char *arch = config->arch;
 	char *server;
diff -Naur pacman-5.0.2-orig/src/pacman/conf.h pacman-5.0.2/src/pacman/conf.h
--- pacman-5.0.2-orig/src/pacman/conf.h	2017-05-18 07:10:47.000000000 -0400
+++ pacman-5.0.2/src/pacman/conf.h	2017-09-01 19:52:55.705446646 -0300
@@ -66,6 +66,7 @@
 	char *dbpath;
 	char *logfile;
 	char *gpgdir;
+	char *codename;
 	alpm_list_t *hookdirs;
 	alpm_list_t *cachedirs;
 
